### ISSUE TITLE
username already taken check?

### ISSUE THREAD

**Issue by frollard** [Protospace/spaceport#99]
https://github.com/Protospace/spaceport/blob/24a0f37faa05884ec87495b50ba394f5f4c3a105/authserver/auth_functions.py#L309

I presume discourse returns a 400ish error but we're not checking for it before trying to change names; don't want to test on live.

**Comment by tannercollin** [comment]
Yeah Discourse will return an error in this case. But this won't be checked if there's a name collision on sign up (ie. another tanner.collin signs up to Spaceport), which can't happen because Spaceport will prevent signup and we'll cross that bridge when it comes. first.last was kind of a mistake.
